code/game/machinery/reagents/pump.dm

```diff
@@ -75,6 +76,9 @@
 	if(!cell_overlay)
 		cell_overlay = new/icon(icon, "[icon_state]-cell")
 
+	if(!panel)
+		panel = new/icon(icon, "[icon_state]-panel")
```

Comment:
I've tried to review this a couple times but I keep tripping over this.
Icon procs are generally a really bad idea if you're not needing to actually generate an icon. They impose network overhead (updating clients with the new icon) as well as being fairly expensive themselves. I'm really perturbed by how much icon ops are used in Polariscode, it's something that is almost certainly contributing to the client crashes.